internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -2288,3 +2301,43 @@ func (r *AgentServiceConfigReconciler) getImageService(ctx context.Context, log
 	}
 	return imageServiceURL
 }
+
+func (r *AgentServiceConfigReconciler) deployAgentInstallCRDs(secret *corev1.Secret, namespace string, log logrus.FieldLogger) error {
+	// Get in-cluster client
+	kubeClient, err := r.K8sApiExtensionsClientFactory.CreateFromInClusterConfig()
```

Comment:
Why use a different client type here? Is there any reason to not use the same kind we're already using everywhere else in this file?

Reply:
We need specifically the 'ApiextensionsV1' API (for listing/creatind CRDs with ApiextensionsV1().CustomResourceDefinitions()). Which seems separate from  [kubernetes.clientset](https://github.com/openshift/assisted-service/blob/84b83af19531313df8186ea4f075894fb518f9a8/vendor/k8s.io/client-go/kubernetes/clientset.go#L19)

Reply:
I'm not following. controller-runtime gives us a great dynamic client with automatic caching, and it's almost always the client we want to use. CRD is [just another type](https://github.com/kubernetes/apiextensions-apiserver/blob/520f4b4/pkg/apis/apiextensions/v1beta1/types.go#L398) that you can [Get](https://sdk.operatorframework.io/docs/building-operators/golang/references/client/#get) or [List](https://sdk.operatorframework.io/docs/building-operators/golang/references/client/#list) like any other resource. Is there some reason not to use it?

Reply:
Seems ACM is using exactly the same apiextensions client for that: https://github.com/open-cluster-management-io/registration-operator/blob/79d280ad8a82055848a25bc9bcb83717f8db27e9/pkg/helpers/helpers.go#L448
I could indeed try to do the same with the dynamic client.
@mhrivnak WDYT, worth the effort to try that or we should keep follow ACM approach with apiextensionsclient?

Reply:
Is there some advantage to using that separate client? Generally it seems simpler to just one client for everything unless there's a compelling reason to bring in a new one.